tests/backup/backup_psa_test.go

```diff
@@ -1,30 +1,590 @@
+	"strings"
+	"sync"
+	"time"
+
 	. "github.com/onsi/ginkgo/v2"
+	"github.com/pborman/uuid"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/portworx/torpedo/drivers/backup"
+	"github.com/portworx/torpedo/drivers/scheduler"
 	"github.com/portworx/torpedo/pkg/log"
 	. "github.com/portworx/torpedo/tests"
 )
 
-// This is a dummy PSA testcase to validate the PSA related methods for RKE
-var _ = Describe("{DummyPSATestcase}", Label(TestCaseLabelsMap[DummyPSATestcase]...), func() {
+var _ = Describe("{EnableNsAndClusterLevelPSAWithBackupAndRestore}", Label(TestCaseLabelsMap[EnableNsAndClusterLevelPSAWithBackupAndRestore]...), func() {
+	var (
+		backupNames          []string
+		restoreNames         []string
+		appList              = Inst().AppList
+		scheduledAppContexts []*scheduler.Context
+		bkpNamespaces        []string
+		label                map[string]string
+		preRuleNameList      []string
+		postRuleNameList     []string
+		providers            []string
+		sourceScNameList     []string
+		cloudCredName        string
+		cloudCredUID         string
+		backupLocationUID    string
+		backupLocationMap    map[string]string
+		sourceClusterUid     string
+		scName               string
+		params               map[string]string
+		backupNSMap          map[string]string
+		controlChannel       chan string
+		errorGroup           *errgroup.Group
+		backupNamesAllNs     []string
+	)
+	storageClassMapping := make(map[string]string)
+	namespaceMapping := make(map[string]string)
+	AppContextsMapping := make(map[string]*scheduler.Context)
+	providers = GetBackupProviders()
+	scheduledAppContexts = make([]*scheduler.Context, 0)
+	bkpNamespaces = make([]string, 0)
+	preRuleNameList = make([]string, 0)
+	postRuleNameList = make([]string, 0)
+	originalList := Inst().AppList
+	label = make(map[string]string)
+	backupLocationMap = make(map[string]string)
+	psaFlag := false
+	backupNSMap = make(map[string]string)
+
 	JustBeforeEach(func() {
-		log.InfoD("No pre-configuration needed")
+		// TODO: Need to update testcase ID
+		StartPxBackupTorpedoTest("EnableNsAndClusterLevelPSAWithBackupAndRestore", "Enable Namespace and cluster level PSA with Backup and Restore", nil, 0, Kshithijiyer, Q2FY25)
+
+		log.InfoD("Deploy applications")
+		scheduledAppContexts = make([]*scheduler.Context, 0)
+		psaApp := make([]string, 0)
+		for _, psalevel := range []string{"restricted", "baseline", "privileged"} {
+			if psalevel == "restricted" {
+				appList := Inst().AppList
+				log.InfoD("The app list at the start of the testcase is %v", Inst().AppList)
+				for _, app := range appList {
+					psaApp = append(psaApp, PSAAppMap[app])
+				}
+				log.Infof("The PSA app list is %v", psaApp)
+				Inst().AppList = psaApp
+			}
+			label["pod-security.kubernetes.io/enforce"] = psalevel
+
+			for i := 0; i < Inst().GlobalScaleFactor; i++ {
+				taskName := fmt.Sprintf("%s-%d-%s-%d", TaskNamePrefix, 0, psalevel, i)
+				namespace := fmt.Sprintf("%s-%d", psalevel, i)
```

Comment:
can we add a random string , it will be help full in case of re-run

Reply:
`DestroyAppsWithData()` will run after the testcase the namespaces will get cleaned up.